consensus/propagation/have_wants.go

```diff
@@ -32,8 +34,8 @@ func (blockProp *Reactor) handleHaves(peer p2p.ID, haves *proptypes.HaveParts, _
 
 	_, parts, fullReqs, has := blockProp.getAllState(height, round, false)
 	if !has {
-		// TODO disconnect from the peer
-		// blockProp.Logger.Debug("received part state for unknown proposal", "peer", peer, "height", height, "round", round)
+		blockProp.Logger.Error("received have part for unknown proposal", "peer", peer, "height", height, "round", round)
+		blockProp.Switch.StopPeerForError(blockProp.getPeer(peer).peer, errors.New("received part for unknown proposal"))
 		return
```

Comment:
I'm not 100% certain if we should add this one yet due to pruning

Reply:
05b0056448553ddddedf781aaf34074e3347f21d